scripts/e2e-test.sh

```diff
@@ -87,19 +87,53 @@ kubectl create namespace eunomia-hello-world-yaml-demo
 
 kubectl apply -f examples/hello-world-yaml/eunomia-runner-sa.yaml -n eunomia-hello-world-yaml-demo
 
-kubectl apply -f examples/hello-world-yaml/cr/hello-world-cr1.yaml -n eunomia-hello-world-yaml-demo
-
-timeout=30
-
-while ((--timeout)) && [[ "$(kubectl get po -n eunomia-hello-world-yaml-demo -l name=hello-world -o=jsonpath="{range .items[*]}{.status.phase}{'\n'}{end}")" != "Running"  ]];do
-  echo "waiting for hello-world deployment: remaining $timeout sec..."
-  sleep 1
-done
-if [[ $timeout == 0 ]]; then
-  echo "Example CR1 Test FAILED"
-  exit 1
-fi
-echo "Example CR1 Test Passed"
-
+timeout=45
+
+#Testing hello-workld-yaml-cr1
+hello_world_yaml_cr_1() {
+  kubectl apply -f examples/hello-world-yaml/cr/hello-world-cr1.yaml -n eunomia-hello-world-yaml-demo
+  while ((--timeout)) && [[ "$(kubectl get po -n eunomia-hello-world-yaml-demo -l name=hello-world -o=jsonpath="{range .items[*]}{.status.phase}{'\n'}{end}")" != "Running"  ]];do
+    echo "waiting for hello-world-yaml-cr1 deployment: remaining $timeout sec..."
+    sleep 2
+  done
+  if [[ $timeout == 0 ]]; then
+    echo "Example hello-world-yaml-cr1 Test FAILED"
+    exit 1
+  fi
+  echo "Example hello-world-yaml-cr1 Test Passed"
+}
+
+#Testing hello_world_yaml_cr_2
+hello_world_yaml_cr_2() {
+  kubectl apply -f examples/hello-world-yaml/cr/hello-world-cr2.yaml -n eunomia-hello-world-yaml-demo
+  while ((--timeout)) && [[ "$(kubectl get replicaset -n eunomia-hello-world-yaml-demo -l name=hello-world -o=jsonpath="{range .items[*]}{.status.readyReplicas}{'\n'}{end}")" != "3"  ]];do
+    echo "waiting for hello-world-yaml-cr1 deployment: remaining $timeout sec..."
+    sleep 2
+  done
+  if [[ $timeout == 0 ]]; then
+    echo "Example hello-world-yaml-cr2 Test FAILED"
+    exit 1
+  fi
+  echo "Example hello-world-yaml-cr2 Test Passed"
+}
+
+#Testing hello_world_yaml_cr_3
+hello_world_yaml_cr_3() {
+  kubectl apply -f examples/hello-world-yaml/cr/hello-world-cr3.yaml -n eunomia-hello-world-yaml-demo
+  while ((--timeout)) && [[ "$(kubectl get deployment -n eunomia-hello-world-yaml-demo -o=jsonpath="{.items[0].metadata.generation}{'\n'}")" != "3"  ]];do
```

Comment:
why `.items[0]`? previous checks have `{range .items[*]}`, why different here?

Reply:
Both works , Will keep unique .

Reply:
I don't know kubectl very well: can you help me understand why cr1 and cr2 use `range` with `items[*]`, but cr3 uses `items[*]` without `range`?

Reply:
It's just used to filter json output . 
cr3  passing values are available in 2 places  in json output, so i used first field. 
Updated now for keeping unique.